pkg/submitter/submitter.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package submitter
+
+import (
+	"fmt"
+	"sort"
+
+	"sqlflow.org/sqlflow/pkg/database"
+	"sqlflow.org/sqlflow/pkg/pipe"
+	"sqlflow.org/sqlflow/pkg/sql/ir"
+
+	pb "sqlflow.org/sqlflow/pkg/proto"
+)
+
+var submitterRegistry = make(map[string]Submitter)
+
+// Submitter submites the Python program by code generator: `pkg/sql/codegen`
+type Submitter interface {
```

Comment:
Since this interface implements `ir.Executor`, it seems its name should be `SubmitterExecutor`, or, simply `Executor`? Also, a *submitter* is a Python program converted from a SQL statement; it doesn't make sense for a Go interface being named *submitter*.

Reply:
I agree with rename to `Executor` and we can move them to `pkg/step` package.

In another, the current implementation, `Submitter` interface embedded with`ir.Executor`, `pai_submitter` implement `Submitter` interface, I think we can move `Executor` from `ir` to `step` package, and add `PAIExecutor`, `AlisaExecutor` to implement the `Executor` interface.

Reply:
Yes. I agree.

Reply:
Or, maybe `paiExecutor`, `alisaExecutor`, and `NewExecutor(submitter_type string)`.